src/radix_tree.cpp

```diff
@@ -0,0 +1,602 @@
+/*
+    Copyright (c) 2018 Contributors as noted in the AUTHORS file
+
+    This file is part of libzmq, the ZeroMQ core engine in C++.
+
+    libzmq is free software; you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    As a special exception, the Contributors give you permission to link
+    this library with independent modules to produce an executable,
+    regardless of the license terms of these independent modules, and to
+    copy and distribute the resulting executable under terms of your choice,
+    provided that you also meet, for each linked independent module, the
+    terms and conditions of the license of that module. An independent
+    module is a module which is not derived from or based on this library.
+    If you modify this library, you must extend this exception to your
+    version of the library.
+
+    libzmq is distributed in the hope that it will be useful, but WITHOUT
+    ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+    FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
+    License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "precompiled.hpp"
+#include "macros.hpp"
+#include "err.hpp"
+#include "radix_tree.hpp"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+node::node(unsigned char* data)
+    : data_(data)
+{}
+
+inline
```

Comment:
It doesn't make sense to declare the methods inline in the cpp file. Either move them to the header file, or remove the inline declaration.

Reply:
Thanks, moved the inline specifiers to the header file. GCC doesn't inline these functions if we remove the specifier; Clang doesn't have this problem, though.